snaps/snapshot.go

```diff
@@ -136,14 +174,17 @@ func updateSnapshot(testID, snapshot, snapPath string) error {
 Returns the dir for snapshots [where the tests run + /snapsDirName]
 and the name [dir + /snapsDirName + /<test-name>.snapsExtName]
 */
-func snapDirAndName() (dir, name string) {
+func snapDirAndName(c *config) (string, string) {
 	callerPath := baseCaller(2)
 	base := filepath.Base(callerPath)
+	dir := filepath.Join(filepath.Dir(callerPath), c.snapsDir)
```

Comment:
For my use case I'd like to specify the directory in full, i.e. not to be influenced by the `callerPath`. There are two cases I encountered for this, one is when ussing go-snaps in a goroutine, the `base` is `$GOROOT/src/runtime`; and the second when I'm using a testing framework that runs a single go test -- in my case [godog](https://github.com/cucumber/godog/), the `base` is `$GOPATH/pkg/mod/github.com/cucumber/godog@v0.12.6`.

I'd prefer if the given `snapsDir` was used as is, perhaps if absolute path was given?
```suggestion
	var dir string
	if filepath.IsAbs(c.snapsDir) {
		dir = c.snapsDir
	} else {
		dir = filepath.Join(filepath.Dir(callerPath), c.snapsDir)
	}
```

Reply:
Hey 👋  Thanks a lot for the feedback and using `go-snaps`. The issue with the file at least for goroutines was fixed  https://github.com/gkampitakis/go-snaps/pull/44. I was able to verify this should not be an issue from `v0.3.7`.

As for providing the directory in full I thought about it but I was a bit hesitant on how it would work with the `Clean`. I like idea of the absolute and I ll play a bit with it as I can see people wanting to use it.

As for the godog unfortunately I am not really familiar with it. I would appreciate if you can share a snippet so I can potentially try and support it regardless of the `snapsDir` configuration.

Reply:
Hello, thanks for considering this!

I've created [an example](https://github.com/zregvart/godog-and-snaps) if I run `go test example_test` this is what I get:

```
go test example_test.go 
Feature: example feature

  Scenario: example scenario     # features/example.feature:3
    Then expect snapshot "hello" # example_test.go:40 -> command-line-arguments.verifySnapshot
    mkdir /home/zregvart/.asdf/installs/golang/1.19.5/packages/pkg/mod/github.com/cucumber/godog@v0.12.6/__snapshots__: permission denied

--- Failed steps:

  Scenario: example scenario # features/example.feature:3
    Then expect snapshot "hello" # features/example.feature:4
      Error: mkdir /home/zregvart/.asdf/installs/golang/1.19.5/packages/pkg/mod/github.com/cucumber/godog@v0.12.6/__snapshots__: permission denied


1 scenarios (1 failed)
1 steps (1 failed)
217.55µs
--- FAIL: TestFeatures (0.00s)
    --- FAIL: TestFeatures/example_scenario (0.00s)
        suite.go:451: mkdir /home/zregvart/.asdf/installs/golang/1.19.5/packages/pkg/mod/github.com/cucumber/godog@v0.12.6/__snapshots__: permission denied
    example_test.go:61: non-zero status returned, failed to run feature tests
FAIL
FAIL	command-line-arguments	0.007s
FAIL
```

It seems to me supporting godog, and in general other similar testing frameworks that abstract away the testing framework might be a challenge -- so thank you for taking an interest and investing time into this.

For my use case, I've configured to use the version from this branch and I've resorted to using [tricks](https://github.com/zregvart/ec-cli/blob/6fed7fd79eea75a843a7adbb501a8f427a0e3ccb/acceptance/snaps/snaps.go#L71-L93) to point to the directory where I want the snapshots to be created.